integration-libs/opf/base/components/opf-quick-buy/opf-quick-buy.service.spec.ts

```diff
@@ -4,4 +4,225 @@
+
+      service.getPaymentGatewayConfiguration().subscribe((result) => {
+        expect(result).toBeUndefined();
+      });
+    });
+  });
+
+  describe('isQuickBuyProviderEnabled', () => {
+    const provider = OpfProviderType.APPLE_PAY;
+
+    it('should return true when the provider is enabled', () => {
+      const activeConfiguration = {
+        digitalWalletQuickBuy: [
+          { provider: OpfProviderType.APPLE_PAY, enabled: true },
+        ],
+      };
+
+      const result = service.isQuickBuyProviderEnabled(
+        provider,
+        activeConfiguration
+      );
+      expect(result).toBeTruthy();
+    });
+
+    it('should return false when the provider is disabled', () => {
+      const activeConfiguration = {
+        digitalWalletQuickBuy: [
+          { provider: OpfProviderType.APPLE_PAY, enabled: false },
+        ],
+      };
+
+      const result = service.isQuickBuyProviderEnabled(
+        provider,
+        activeConfiguration
+      );
+      expect(result).toBeFalsy();
+    });
+
+    it('should return false when the provider is not found', () => {
+      const activeConfiguration = {
+        digitalWalletQuickBuy: [
+          { provider: 'otherProvider' as any, enabled: true },
+        ],
+      };
+
+      const result = service.isQuickBuyProviderEnabled(
+        provider,
+        activeConfiguration
+      );
+      expect(result).toBeFalsy();
+    });
+
+    it('should return false when activeConfiguration is null', () => {
+      const result = service.isQuickBuyProviderEnabled(provider, null as any);
+      expect(result).toBeFalsy();
+    });
+
+    it('should return false when activeConfiguration is undefined', () => {
+      const result = service.isQuickBuyProviderEnabled(
+        provider,
+        undefined as any
+      );
+      expect(result).toBeFalsy();
+    });
+
+    it('should return false when digitalWalletQuickBuy is null or empty', () => {
+      const provider = OpfProviderType.APPLE_PAY;
+      const activeConfiguration = {
+        digitalWalletQuickBuy: null as any,
+      };
+
+      const result = service.isQuickBuyProviderEnabled(
+        provider,
+        activeConfiguration
+      );
+      expect(result).toBeFalsy();
+    });
+  });
+
+  describe('isUserGuestOrLoggedIn', () => {
+    it('should return true if the user is logged in', () => {
+      baseSiteServiceMock.get.and.returnValue(
+        of({ baseStore: { paymentProvider: 'providerWithGuestSupport' } })
+      );
+      authServiceMock.isUserLoggedIn.and.returnValue(of(true));
+
+      service.isUserGuestOrLoggedIn().subscribe((result) => {
+        expect(result).toBeTruthy();
+      });
+    });
+
+    it('should return true if the user is a guest and guest checkout is supported', () => {
+      baseSiteServiceMock.get.and.returnValue(
+        of({ baseStore: { paymentProvider: 'providerWithGuestSupport' } })
+      );
+
+      checkoutConfigMock.checkout.and.returnValue({
```

Comment:
small thing: I think it can be removed as defined on line 195. Moreover, following 'it' block does not have it.

Reply:
Right, that was a leftover. Thanks for catching it. Removed from code.